src/main/java/juke/input/Parser.java

```diff
@@ -0,0 +1,154 @@
+package juke.input;
+
+import juke.exception.ParserException;
+import juke.main.Constants;
+
+public class Parser {
+    private static String textInput;
+    private static String inputCommand = "invalid";
+    private static String taskName;
+    private static String taskNumber;
+    private static int taskIndex = -1;
+    private static int spaceIndex = -1;
+    private static int slashIndex = -1;
+    private static int slashIndex2 = -1;
+    private static String deadline;
+    private static String from;
+    private static String to;
```

Comment:
If any of these are not going to change in the duration of the program, you can change them to constants `static final` in `SCREAMING_SNAKE_CASE`